packages/react-components/src/components/InAppMessages/InAppMessage.tsx

```diff
@@ -0,0 +1,104 @@
+import * as React from 'react';
+import cx from 'clsx';
+
+import {
+  InAppMessageHeader,
+  InAppMessageHeaderProps,
+} from './InAppMessageHeader';
+import { InAppMessageImage, InAppMessageImageProps } from './InAppMessageImage';
+import {
+  InAppMessageFooter,
+  InAppMessageFooterProps,
+} from './InAppMessageFooter';
+import { KeyCodes } from '../../utils/keyCodes';
+
+import styles from './InAppMessage.module.scss';
+
+const baseClass = 'in-app-message';
+
+export interface InAppMessageProps {
+  className?: string;
+  image?: InAppMessageImageProps;
+  header?: InAppMessageHeaderProps;
+  footer?: InAppMessageFooterProps;
+  closeOnEscPress?: boolean;
+  onClose: () => void;
+}
+
+export const InAppMessage: React.FC<InAppMessageProps> = ({
+  children,
+  className,
+  closeOnEscPress = true,
+  image,
+  header,
+  footer,
+  onClose,
+}) => {
+  React.useEffect(() => {
+    const onDocumentClick = (e: MouseEvent) => {
+      if (e.target === e.currentTarget) {
+        handleCloseInAppMessage();
+      }
+    };
```

Comment:
Why need a document click handler if there is already a backdrop/overly click handler? 🤔